HTTP_HEADER_FORMAT.md

```diff
@@ -0,0 +1,88 @@
+# Trace Context HTTP Header Format
+
+Date: 31/03/2017
+
+A trace context header is used to pass trace context information across systems
+for a HTTP request. Our goal is to share this with the community so that various
+tracing and diagnostics products can operate together, and so that services can
+pass context through them, even if they're not being traced (useful for load
+balancers, etc.)
+
+# Format
+
+## Header name
+
+`Trace-Context`
+
+## Field value
+
+`base16(<version><version_format>)`
+
+The value will be US-ASCII encoded (which is UTF-8 compliant).
+
+### Version
+
+Is a 1-byte representing a uint8 value.
+
+### Version = 0
+
+#### Format
+
+`<trace-id><span-id><trace-options>`
+
+All fields are required.
+
+#### Trace-id
+
+Is the ID of the whole trace forest. It is represented as a 16-bytes array,
+e.g., `0x4bf92f3577b34da6a3ce929d0e0e4736`. All bytes 0 is considered invalid.
+
+Implementation may decide to completely ignore the trace-context if the trace-id
+is invalid.
+
+#### Span-id
+
+Is the ID of the caller span (parent). It is represented as a 8-bytes array,
+e.g., `0x00f067aa0ba902b7`. All bytes 0 is considered invalid.
+
+Implementation may decide to completely ignore the trace-context if the span-id
+is invalid.
+
+#### Trace-options
+
+Controls tracing options such as sampling, trace level etc. It is a 4-bytes
+representing a 32-bit unsigned integer in little-endian order. The least
```

Comment:
ps to elaborate on this, there's a couple things in play. On one hand, if you have only one flag, the endianness isn't an issue. Most users have sampled set as a single bit, and that's the only flag used in zipkin routinely.

On the other hand, the platform default encoding of java as big-endian results in many not being aware of endian-ness in general. Mistakes coding anything binary have been difficult in zipkin, resulting in json actually. While the case endianness is "little" elsewhere, anectodally I've noticed those who have little endian platforms already know how to do big endian. Main thing is that they get annoyed when endian isn't documented. Ex when folks were doing thrift encoding I remember someone rather annoyed that thrift tbinary format was big endian without saying so (because it was implicitly that in java).

food for thought